app/src/main/java/com/example/tasky/authentication/data/local/UserCacheImpl.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.tasky.authentication.data.local
+
+import android.content.SharedPreferences
+import com.example.tasky.authentication.domain.UserCache
+import com.example.tasky.authentication.domain.model.User
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class UserCacheImpl @Inject constructor(
+    private val sharedPreferences: SharedPreferences
+) : UserCache {
+
+    private val moshi = Moshi.Builder()
+        .add(KotlinJsonAdapterFactory())
+        .build()
+    private val adapter = moshi.adapter(User::class.java)
+    override fun saveUser(user: User) {
+        sharedPreferences.edit()
+            .putString(userKey, adapter.toJson(user))
```

Comment:
Its fine to save as json

But, you probably wont use the whole User object everytime you get the info

Ex: jwtinterceptor will just use the jwt. Not the id or fullname

Personally I save 3 different keys (token, fullname, user id) and save each value to each key